### PR TITLE
Problem: released goleveldb binaries has external libc dependencies

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #464
Solution:
- build linux and x86_64-darwin goleveldb release binaries using nix too.
- upload checksums for nix built binaries.

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: default.nix
Comment (yihuang) [inline default.nix:54]: https://github.com/NixOS/nixpkgs/blob/master/lib/licenses.nix#L95
Apache License 2.0

File: flake.nix
Comment (yihuang) [inline flake.nix:62]: don't follow symbolic links.
don't leak the building user in the file attributes.
keep the normal file permissions.
follow hard links.

File: flake.nix
Comment (yihuang) [inline flake.nix:63]: keep gzip dependency explicit for better reproducibility.

File: default.nix
Comment (yihuang) [inline default.nix:13]: see: https://github.com/3noch/nix-bundle-exe/issues/14

File: flake.lock
Comment (yihuang) [inline flake.lock:44]: TODO, update after upstream PR merged.

File: .github/workflows/release.yml
Comment (yihuang) [inline .github/workflows/release.yml:57]:
```suggestion
          sha256sum *.tar.gz > "checksums-$PLATFORM.txt"
```

File: .goreleaser.yml
Comment (yihuang) [inline .goreleaser.yml:96]:
```suggestion
  name_template: 'checksums-others.txt'
```

File: flake.nix
Comment (yihuang) [inline flake.nix:63]: ref the nix gzip explicitly for better reproducibility.

File: flake.nix
Comment (tomtau) [reply flake.nix:62]: maybe good to put this as a comment directly in the source code?

**Comment by yihuang** [comment]
https://github.com/yihuang/cronos/releases/tag/v0.0.38-test
test run result.

File: flake.nix
Comment (yihuang) [reply flake.nix:64]: sounds good, done.

**Review by tomtau — APPROVED** [review]
should this undergo QA testing that e.g. ledger / usb is still working with these new bundled binaries on different platforms?

**Comment by yihuang** [comment]
> should this undergo QA testing that e.g. ledger / usb is still working with these new bundled binaries on different platforms?

sounds good. loop in @macong-cdc

**Comment by tomtau** [comment]
oh ok, in that case, one can keep using the old binary for CLI

**Comment by yihuang** [comment]
> oh ok, in that case, one can keep using the old binary for CLI

I'll merge this for now, if the test failed, we'll fix in another PR.

**Pull request merged**
